ratelimit/redis.go

```diff
@@ -120,78 +117,70 @@ func (c *clusterLimitRedis) startSpan(ctx context.Context, spanName string) func
 //
 // If a context is provided, it uses it for creating an OpenTracing span.
 func (c *clusterLimitRedis) AllowContext(ctx context.Context, clearText string) bool {
-	s := getHashedKey(clearText)
 	c.metrics.IncCounter(redisMetricsPrefix + "total")
+	now := time.Now()
+	finishSpan := c.startSpan(ctx, allowSpanName)
+
+	allow, err := c.allowContext(ctx, clearText)
+
+	finishSpan(err != nil)
+	queryFailure := err != nil
+	c.measureQuery(allowMetricsFormat, allowMetricsFormatWithGroup, &queryFailure, now)
+
+	if err != nil {
```

Comment:
not sure if this makes sense in details, but what if we would use `queryFailure` in all of the `err != nil` use

Reply:
Looks better indeed, fixed.